nixos/modules/services/misc/sourcehut/builds.nix

```diff
@@ -0,0 +1,166 @@
+    user = mkOption {
+      type = types.str;
+      default = "buildsrht";
+      description = ''
+        User for builds.sr.ht.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 5002;
+      description = ''
+      '';
+    };
+
+    database = mkOption {
+      type = types.str;
+      default = "builds.sr.ht";
+      description = ''
+        PostgreSQL database name for builds.sr.ht.
+      '';
+    };
+
+    statePath = mkOption {
+      type = types.path;
+      default = "${cfg.statePath}/buildsrht";
+      description = ''
+        State path for builds.sr.ht.
+      '';
+    };
+
+    workers = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Run workers for builds.sr.ht.
+      '';
+    };
+  };
+
+  config = with scfg; lib.mkIf (cfg.enable && elem "builds" cfg.services) {
+    users = {
+      users = { "${user}" = {
+          group = user;
+          extraGroups = if cfg.builds.workers then ["docker"] else [];
+          description = "builds.sr.ht user"; };
+      };
+
+      groups = {"${user}" = {};};
+    };
+
+    services.postgresql = {
+      authentication = ''
+        local ${database} ${user} trust
+      '';
+      ensureDatabases = [ database ];
+      ensureUsers = [
+        { name = user;
+          ensurePermissions = { "DATABASE \"${database}\"" = "ALL PRIVILEGES"; }; }
+      ];
+    };
+
+    systemd = {
+      tmpfiles.rules = [
+        "d ${statePath} 0750 ${user} ${user} -"
+      ] ++ (lib.optional cfg.builds.workers "d /var/lib/images 0775 ${user} ${user} - -");
+
+      services = {
+        buildsrht = import ./service.nix { inherit config pkgs lib; } scfg drv iniKey {
+          after = [ "postgresql.service" "network.target" ];
+          requires = [ "postgresql.service" ];
+          wantedBy = [ "multi-user.target" ];
+
+          description = "builds.sr.ht website service";
+
+          serviceConfig.ExecStart = "${cfg.python}/bin/gunicorn ${drv.pname}.app:app -b ${cfg.address}:${toString port}";
+
+    # Hack to bypass this hack: https://git.sr.ht/~sircmpwn/core.sr.ht/tree/master/item/srht-update-profiles#L6
+        } // { preStart = " ";};
+
+        buildsrht-worker = {
+          enable = cfg.builds.workers;
+          after = [ "postgresql.service" "network.target" ];
+          requires = [ "postgresql.service" ];
+          wantedBy = [ "multi-user.target" ];
+          description = "builds.sr.ht worker service";
+          path = [ pkgs.docker pkgs.openssh pkgs.rsync ];
+          serviceConfig = {
+            Type = "simple";
+            User = user;
+            Group = "nginx";
+            Restart = "always";
+          };
+          serviceConfig.ExecStart = "${pkgs.sourcehut.buildsrht}/bin/builds.sr.ht-worker";
+
+          # TODO: properly create and register build images and have a configuration to add more
+          preStart = let
+              pkgs_unstable= builtins.fetchGit {
```

Comment:
-1 on this. Definitely not a fan of having to clone yet another Nixpkgs instance, especially since it's so heavy-weight these days.

Reply:
Should we pin some to 20.09 and 20.03? And one to the nixos-unstable (or just one that is the inherited nixpkgs under evaluation?)

This should also be exposed via an option setting of some kind rather than hardcoded in this fashion.

Reply:
oh, and just use the nix correct path rather than `/var/lib/images`...

Reply:
I'd definitely say make it an option. Note that modules have access to `pkgs`, so we don't need to import packages *again* (as this will get stale very quickly).